### DISCUSSION TITLE
it seems that memory watermark did not work

### DISCUSSION THREAD

**Discussion by akaxb** [rabbitmq/rabbitmq-server#9189] [Other]
### Describe the bug

server version: 3.6.6
watermark configuration：Memory limit set to 6354MB of 15886MB total(40%).
describe：
The mq process was killed by the Linux OOM, and the system log records that during the kill, the memory usage of mq is as follows:
**total-vm:17600196KB,anon-rss:15291812KB**

_My question is, since I have set the water level of the memory(40%), why does it appear that this control has failed_

Under what circumstances will memory threshold fail?

- this is system log:
![img_v2_ad72658f-e173-4c87-9570-2688ef51fcbg](https://github.com/rabbitmq/rabbitmq-server/assets/6300834/31f2a099-42d4-47b4-b8af-b0308953c04c)
- Before OOM, there were no particularly obvious anomalies observed in Grafana:
<img width="1324" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/6300834/dfd927e8-dca1-4664-a8b9-2a6a0b83e4ca">
- the vhost messages  in Grafana:
<img width="664" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/6300834/9206f72d-991f-48bd-923d-f7c4501dc35e">

### Reproduction steps

nothing

### Expected behavior

MQ uses up to 40% of the host's memory

**Comment by michaelklishin** [comment]
RabbitMQ [3.6 has reached EOL in May 2018](https://rabbitmq.com/versions.html). [Please upgrade](https://rabbitmq.com/upgrade.html), that's the only recommendation we have.

The watermark is not a hard resource limit. It is used for nodes to decide [when it's time to block publishers](https://rabbitmq.com/alarms.html), not when the node will stop allocating memory.

Start with [understanding what consumes memory on the node](https://rabbitmq.com/memory-use.html).

Since 3.6, a lot has changed in the area of memory management: [streams](https://rabbitmq.com/streams.html), quorum queues and [CQv2 classic queues](https://blog.rabbitmq.com/posts/2023/05/rabbitmq-3.12-performance-improvements/) keep significantly less data in memory. Please pick the option that fits your needs.

There's also this [curious case of CentOS 9 default. changes](https://blog.rabbitmq.com/posts/2022/08/high-initial-memory-consumption-of-rabbitmq-nodes-on-centos-stream-9/) that can and will affect RabbitMQ nodes.
